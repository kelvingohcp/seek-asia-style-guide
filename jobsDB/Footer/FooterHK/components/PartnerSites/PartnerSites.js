import React from 'react';
import PropTypes from 'prop-types';

import styles from './PartnerSites.less';
import { ListItem, DropdownList } from 'seek-asia-style-guide/react';

const PartnerSites = ({ messages, externalLink }) => {
  return (
    <DropdownList
      value={messages['FooterHK.partners']}
      noShadow
      className={styles.content}
      semiStrong
      whistling
      chevronAlignment='center'>

      {externalLink(
        <ListItem value='BabaJob (India)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.BabaJobLink']
      )}
      {externalLink(
        <ListItem value='Bdjobs (Bandladesh)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.BdjobsLink']
      )}
      {externalLink(
        <ListItem value='Brighter Monday (East Africa)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.BrighterMondayLink']
      )}
      {externalLink(
        <ListItem value='Catho (Brazil)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.CathoLink']
      )}
      {externalLink(
        <ListItem value='Jobberman (West Africa)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.JobbermanLink']
      )}
      {externalLink(
        <ListItem value='JobStreet (S.E. Asia)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.JobStreetLink']
      )}
      {externalLink(
        <ListItem value='Jora (Hong Kong)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.JoraLink']
      )}
      {externalLink(
        <ListItem value='Manager (Brazil)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.ManagerLink']
      )}
      {externalLink(
        <ListItem value='OCC Mundial (Mexico)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.OCCLink']
      )}
      {externalLink(
        <ListItem value='Seaman Jobsite (Philippines)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.SeamanJobsiteLink']
      )}
      {externalLink(
        <ListItem value='SEEK (Australia)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.SEEKLink']
      )}
      {externalLink(
        <ListItem value='Work Abroad (Philippines)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.WorkAbroadLink']
      )}
      {externalLink(
        <ListItem value='Workana' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.WorkanaLink']
      )}
      {externalLink(
        <ListItem value='Zhaopin (China)' className={styles.partners} descriptionProps={{ semiStrong: true, whistling: true }} />,
        messages['FooterHK.ZhaopinLink']
      )}
    </DropdownList>
  );
};

PartnerSites.propTypes = {
  messages: PropTypes.object.isRequired,
  externalLink: PropTypes.func.isRequired
};

export default PartnerSites;
