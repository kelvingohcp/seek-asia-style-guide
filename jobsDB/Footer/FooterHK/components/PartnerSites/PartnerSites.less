@import (reference) "~seek-asia-style-guide/theme";
@import "../UpperFooter/UpperFooter.less";

.partners:extend(.contentBase) {
  padding-bottom: 0;
}
