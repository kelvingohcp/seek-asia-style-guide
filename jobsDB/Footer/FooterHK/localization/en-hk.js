export default {
  'FooterHK.cmsLink': '/en-hk',
  'FooterHK.cfsLink': '/hk/en',
  'FooterHK.titleAboutJobsDB': 'About JobsDB',
  'FooterHK.titleJobSeeker': 'Job Seekers',
  'FooterHK.titleEmployer': 'Employers',
  'FooterHK.titleConnect': 'Connect',
  'FooterHK.copyright': 'Â© jobsDB. All Rights Reserved',
  'FooterHK.siteMapLink': '/sitemap',
  'FooterHK.siteMapLinkText': 'Site Map',
  'FooterHK.termsLink': '/pages/terms/terms-conditions',
  'FooterHK.termsLinkText': 'Terms & Conditions',
  'FooterHK.privacyLink': '/pages/terms/privacy-policy',
  'FooterHK.privacyLinkText': 'Privacy Statement',
  'FooterHK.aboutJobsDB': 'About jobsDB',
  'FooterHK.aboutJobsDBLink': '/pages/aboutus',
  'FooterHK.faq': 'FAQ',
  'FooterHK.faqLink': '/StaticContentFullContainer/AboutUS/faq/index.htm',
  'FooterHK.career': 'Career @ jobsDB',
  'FooterHK.careerLink': '/Search/FindJobs?AccNums=1&JSRV=1&page=1',
  'FooterHK.contactUs': 'Contact us',
  'FooterHK.contactUsLink': '/pages/regional-coverage',
  'FooterHK.partners': 'International partners',
  'FooterHK.browseJobs': 'Browse jobs',
  'FooterHK.browseJobsLink': '/browse',
  'FooterHK.resumes': 'Post resumes',
  'FooterHK.resumesLink': '/Resumes/ResumeIndex',
  'FooterHK.jobAlerts': 'Job Alerts',
  'FooterHK.jobAlertsLink': '/JobAlert/JobAlertCreation',
  'FooterHK.careerInsights': 'Career Insights',
  'FooterHK.postJob': 'Post a job',
  'FooterHK.postJobLink': '/cms/employer/product-services/job-ads/',
  'FooterHK.searchCandidates': 'Search candidates',
  'FooterHK.searchCandidatesLink': '/cms/employer/product-services/talent-search/',
  'FooterHK.advertise': 'Advertise with us',
  'FooterHK.advertiseLink': '/cms/employer/product-and-services/',
  'FooterHK.myJobsDBTitle': 'My jobsDB',
  'FooterHK.myJobsDBUrl': '/MyJobsDB',
  'FooterHK.careerInsightsTitle': 'Career Insights',
  'FooterHK.careerInsightsUrl': '/career-insights',
  'FooterHK.bestCompanies': 'Best Companies',
  'FooterHK.bestCompaniesLink': '/companies',
  'FooterHK.facebook': 'Facebook',
  'FooterHK.facebookLink': 'https://www.facebook.com/JobsDB.HongKong',
  'FooterHK.twitter': 'Twitter',
  'FooterHK.twitterLink': 'https://twitter.com/HKJobs',
  'FooterHK.appStore': 'jobsDB @ App Store',
  'FooterHK.appStoreLink': 'https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432',
  'FooterHK.googlePlay': 'jobsDB @ Google Play',
  'FooterHK.googlePlayLink': 'https://play.google.com/store/apps/details?id=com.jobsdb',
  'FooterHK.BdjobsLink': 'https://www.bdjobs.com/',
  'FooterHK.BrighterMondayLink': 'https://www.brightermonday.com/',
  'FooterHK.CathoLink': 'https://www.catho.com.br/',
  'FooterHK.JobbermanLink': 'https://www.jobberman.com/',
  'FooterHK.JobStreetLink': 'https://www.jobstreet.com/',
  'FooterHK.JoraLink': 'https://hk.jora.com/',
  'FooterHK.OCCLink': 'https://www.occ.com.mx/',
  'FooterHK.SeamanJobsiteLink': 'https://www.seamanjobsite.com/',
  'FooterHK.SEEKLink': 'https://www.seek.com.au/',
  'FooterHK.WorkAbroadLink': 'https://www.workabroad.ph/',
  'FooterHK.WorkanaLink': 'https://www.workana.com/',
  'FooterHK.ZhaopinLink': 'https://www.zhaopin.com/'
};
