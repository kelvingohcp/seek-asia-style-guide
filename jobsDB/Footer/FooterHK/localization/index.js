import enHkLocalisation from './en-hk';

const localization = {
  'en-hk': enHkLocalisation
};

export const getLocalization = ({ language, country }) => {
  return localization[`${language.toLowerCase()}-${country.toLowerCase()}`] ?
    localization[`${language.toLowerCase()}-${country.toLowerCase()}`] : {};
};

export default localization;
