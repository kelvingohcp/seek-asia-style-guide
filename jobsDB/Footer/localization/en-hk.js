const domain = 'jobsdb.com';
const baseUrl = `https://hk.${domain}`;
const baseUrlWithLanguage = `${baseUrl}/hk/en`;
const wpUrlWithLanguage = `${baseUrl}/en-hk`;
const mbaseUrl = `https://m.${domain}`;
const mUrlWithLanguage = `${mbaseUrl}/en-hk`;

export default {
  'footer.copyright': 'Copyright Â© 1998-{year}, jobsDB.',
  'footer.rightReserved': 'All Rights Reserved.',
  'footer.siteMapLink': `${baseUrlWithLanguage}/sitemap`,
  'footer.siteMapLinkText': 'Site Map',
  'footer.termsLink': `${wpUrlWithLanguage}/pages/terms/terms-conditions`,
  'footer.termsLinkText': 'Terms & Conditions',
  'footer.privacyLink': `${wpUrlWithLanguage}/pages/terms/privacy-policy`,
  'footer.privacyLinkText': 'Privacy Statement',
  'footer.legalLink': `${mUrlWithLanguage}/legal.do`,
  'footer.legalLinkText': 'Legal terms'
};
