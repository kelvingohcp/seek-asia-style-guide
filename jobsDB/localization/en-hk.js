import { FlagHKIcon, FlagIDIcon, FlagSGIcon, FlagTHIcon } from 'seek-asia-style-guide/react';

export default {
  'header.homeTitle': 'Home',
  'header.homeUrl': '/hk',
  'header.mobileLoggedHomeUrl': 'https://m.jobsdb.com/en-hk/jobsForYou.do',
  'header.myJobsDBTitle': 'My jobsDB',
  'header.myJobsDBUrl': '/hk/en/MyJobsDB',
  'header.myJobsDBUrlLoggedIn': '/hk/en/MyJobsDB',
  'header.resourcesTitle': 'Resources',
  'header.resourcesUrl': '/en-hk/career-recruitment-advice/',
  'header.careerInsightsTitle': 'Career Insights',
  'header.careerInsightsUrl': '/hk/en/career-insights',
  'header.employerSiteTitle': 'For Employers',
  'header.employerSiteUrl': '/hk/en/home/employerindex',
  'header.loginTitle': 'Log in',
  'header.loginUrl': '/hk/en/login/jobseekerlogin',
  'header.signupTitle': 'Sign up',
  'header.signupUrl': '/hk/en/jobseeker/registration',
  'header.profileTitle': 'View profile',
  'header.profileUrl': '/hk/en/MyJobsDB/profile',
  'header.invitationTitle': 'Job invitations',
  'header.invitationUrl': '/hk/en/Invitations/InvitationList',
  'header.logoutTitle': 'Log out',
  'header.logoutUrl': 'https://www.thisislogoutlink.test',
  'header.seperatorOr': 'or',
  'header.searchUrl': '/HK/EN/Search/FindJobs?KeyOpt=COMPLEX&JSRV=1&RLRSF=1&JobCat=1&JSSRC=HPSS',
  'menu.jobSeekerHeader': 'JOB SEEKER',
  'menu.jobSeekerProfile': 'PROFILE',
  'menu.settingsHeader': 'SETTINGS',
  'menu.more': 'More',
  'menu.moreHeader': 'MORE',
  'menu.countryAndLanguage': 'Country & Language',
  'menu.backToMenu': 'Back to menu',
  'header.countryList': [ {
    title: 'Hong Kong (English)',
    ItemIcon: FlagHKIcon,
    url: '',
    language: 'en',
    country: 'hk',
    languageDisplay: 'English'
  }, {
    title: 'Indonesia (English)',
    ItemIcon: FlagIDIcon,
    url: '',
    language: 'en',
    country: 'id',
    languageDisplay: 'English'

  }, {
    title: 'Indonesia (Bahasa)',
    ItemIcon: FlagIDIcon,
    url: '',
    language: 'id',
    country: 'id',
    languageDisplay: 'Bahasa'
  }, {
    title: 'Singapore (English)',
    ItemIcon: FlagSGIcon,
    url: '',
    language: 'en',
    country: 'sg',
    languageDisplay: 'English'
  }, {
    title: 'Thailand (English)',
    ItemIcon: FlagTHIcon,
    url: '',
    language: 'en',
    country: 'th',
    languageDisplay: 'English'
  }, {
    title: 'ไทย (ภาษาไทย)',
    ItemIcon: FlagTHIcon,
    url: '',
    language: 'th',
    country: 'th',
    languageDisplay: 'ภาษาไทย'
  } ]
};
