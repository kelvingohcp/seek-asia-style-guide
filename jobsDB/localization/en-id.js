import {
  FlagHKIcon,
  FlagIDIcon,
  FlagSGIcon,
  FlagTHIcon
} from 'seek-asia-style-guide/react';

export default {
  'header.homeTitle': 'Home',
  'header.homeUrl': '/id',
  'header.myJobsDBTitle': 'My jobsDB',
  'header.myJobsDBUrl': '/id/en/MyJobsDB',
  'header.myJobsDBUrlLoggedIn': '/id/en/MyJobsDB',
  'header.resourcesTitle': 'Resources',
  'header.resourcesUrl': '/id-id/career-recruitment-advice/',
  'header.careerInsightsTitle': 'Event',
  'header.careerInsightsUrl': '/id-id/pages/event',
  'header.employerSiteTitle': 'For Employers',
  'header.employerSiteUrl': '/id/en/home/employerindex',
  'header.loginTitle': 'Log in',
  'header.loginUrl': '/id/en/login/jobseekerlogin',
  'header.signupTitle': 'Sign up',
  'header.signupUrl': '/id/en/jobseeker/registration',
  'header.profileTitle': 'View profile',
  'header.profileUrl': '/id/en/MyJobsDB/profile',
  'header.invitationTitle': 'Job invitations',
  'header.invitationUrl': '/id/en/Invitations/InvitationList',
  'header.logoutTitle': 'Log out',
  'header.logoutUrl': 'https://www.thisislogoutlink.test',
  'header.seperatorOr': 'or',
  'header.searchUrl':
    '/ID/EN/Search/FindJobs?KeyOpt=COMPLEX&JSRV=1&RLRSF=1&JobCat=1&JSSRC=HPSS',
  'menu.jobSeekerHeader': 'JOB SEEKER',
  'menu.jobSeekerProfile': 'PROFILE',
  'menu.settingsHeader': 'SETTINGS',
  'menu.more': 'More',
  'menu.moreHeader': 'MORE',
  'menu.countryAndLanguage': 'Country & Language',
  'menu.backToMenu': 'Back to menu',
  'header.countryList': [
    {
      title: 'Hong Kong (English)',
      ItemIcon: FlagHKIcon,
      url: '',
      language: 'en',
      country: 'hk',
      languageDisplay: 'English'
    },
    {
      title: 'Indonesia (English)',
      ItemIcon: FlagIDIcon,
      url: '',
      language: 'en',
      country: 'id',
      languageDisplay: 'English'
    },
    {
      title: 'Indonesia (Bahasa)',
      ItemIcon: FlagIDIcon,
      url: '',
      language: 'id',
      country: 'id',
      languageDisplay: 'Bahasa'
    },
    {
      title: 'Singapore (English)',
      ItemIcon: FlagSGIcon,
      url: '',
      language: 'en',
      country: 'sg',
      languageDisplay: 'English'
    },
    {
      title: 'Thailand (English)',
      ItemIcon: FlagTHIcon,
      url: '',
      language: 'en',
      country: 'th',
      languageDisplay: 'English'
    },
    {
      title: 'ไทย (ภาษาไทย)',
      ItemIcon: FlagTHIcon,
      url: '',
      language: 'th',
      country: 'th',
      languageDisplay: 'ภาษาไทย'
    }
  ]
};
