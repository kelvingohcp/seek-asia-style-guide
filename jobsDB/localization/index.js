import enHkLocalization from './en-hk';
import enSgLocalization from './en-sg';
import enIdLocalization from './en-id';
import idIdLocalization from './id-id';
import enThLocalization from './en-th';
import thThLocalization from './th-th';

const localization = {
  'en-hk': enHkLocalization,
  'en-sg': enSgLocalization,
  'en-id': enIdLocalization,
  'id-id': idIdLocalization,
  'en-th': enThLocalization,
  'th-th': thThLocalization
};

export const getLocalization = ({ language, country }) => {
  return localization[`${language.toLowerCase()}-${country.toLowerCase()}`]
    ? localization[`${language.toLowerCase()}-${country.toLowerCase()}`]
    : {};
};

export default localization;
