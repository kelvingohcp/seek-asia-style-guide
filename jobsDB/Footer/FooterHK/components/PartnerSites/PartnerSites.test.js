import React from 'react';
import { shallow } from 'enzyme';
import PartnerSites from './PartnerSites';
import localization from '../../localization';

describe('PartnerSites:', () => {
  it('should render list with messages and linking function', () => {
    const wrapper = shallow(
      <PartnerSites messages={localization['en-hk']} externalLink={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
