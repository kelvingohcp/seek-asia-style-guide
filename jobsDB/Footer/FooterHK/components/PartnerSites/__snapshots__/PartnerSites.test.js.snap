// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnerSites: should render list with messages and linking function 1`] = `
<DropdownList
  className="content"
  compact={true}
  descriptionProps={
    Object {
      "semiStrong": true,
      "whistling": true,
    }
  }
  noShadow={true}
  value="International partners"
/>
`;
